test/conformance/CMakeLists.txt

```diff
@@ -66,29 +68,50 @@ function(add_conformance_test name)
         unit_tests_helpers)
 
     if(UR_BUILD_ADAPTER_CUDA OR UR_BUILD_ADAPTER_ALL)
-        add_test_adapter(${name} adapter_cuda)
+        add_test_adapter(${name} adapter_cuda CUDA ON)
     endif()
     if(UR_BUILD_ADAPTER_HIP OR UR_BUILD_ADAPTER_ALL)
-        add_test_adapter(${name} adapter_hip)
+        add_test_adapter(${name} adapter_hip HIP ON)
     endif()
     if(UR_BUILD_ADAPTER_L0 OR UR_BUILD_ADAPTER_ALL)
-        add_test_adapter(${name} adapter_level_zero)
+        add_test_adapter(${name} adapter_level_zero LEVEL_ZERO ON)
     endif()
     if(UR_BUILD_ADAPTER_L0_V2)
-        add_test_adapter(${name} adapter_level_zero_v2)
+        add_test_adapter(${name} adapter_level_zero_v2 LEVEL_ZERO ON)
     endif()
     if(UR_BUILD_ADAPTER_OPENCL OR UR_BUILD_ADAPTER_ALL)
-        add_test_adapter(${name} adapter_opencl)
+        add_test_adapter(${name} adapter_opencl OPENCL ON)
     endif()
     if(UR_BUILD_ADAPTER_NATIVE_CPU OR UR_BUILD_ADAPTER_ALL)
-        add_test_adapter(${name} adapter_native_cpu)
+        add_test_adapter(${name} adapter_native_cpu NATIVE_CPU ON)
+    endif()
+
+    if(UR_CONFORMANCE_TEST_LOADER)
+        if(UR_BUILD_ADAPTER_CUDA OR UR_BUILD_ADAPTER_ALL)
+            add_test_adapter(${name} adapter_cuda_loader CUDA OFF)
+        endif()
+        if(UR_BUILD_ADAPTER_HIP OR UR_BUILD_ADAPTER_ALL)
+            add_test_adapter(${name} adapter_hip_loader HIP OFF)
+        endif()
+        if(UR_BUILD_ADAPTER_L0 OR UR_BUILD_ADAPTER_ALL)
+            add_test_adapter(${name} adapter_level_zero_loader LEVEL_ZERO OFF)
+        endif()
+        if(UR_BUILD_ADAPTER_L0_V2)
+            add_test_adapter(${name} adapter_level_zero_v2_loader LEVEL_ZERO OFF)
+        endif()
+        if(UR_BUILD_ADAPTER_OPENCL OR UR_BUILD_ADAPTER_ALL)
+            add_test_adapter(${name} adapter_opencl_loader OPENCL OFF)
+        endif()
+        if(UR_BUILD_ADAPTER_NATIVE_CPU OR UR_BUILD_ADAPTER_ALL)
+            add_test_adapter(${name} adapter_native_cpu_loader NATIVE_CPU OFF)
+        endif()
```

Comment:
It might be good for this stuff to be very explicit but I wonder if there's value in refactoring this into a function (if possible, I'm not super experienced with CMake) as there's a lot of repetition?